src/ConnectionAutomaton.cpp

```diff
@@ -293,6 +310,17 @@ void ConnectionAutomaton::onConnectionFrame(
       }
       return;
     }
+    case FrameType::ERROR: {
+      Frame_ERROR frame;
+      if (frame.deserializeFrom(std::move(payload))) {
+        if (frame.errorCode_ == ErrorCode::CONNECTION_ERROR && !isServer_ && isResumable_ && resumeStatusCallback_) {
+          resumeStatusCallback_->onConnectionError(std::runtime_error(frame.payload_.moveDataToString()));
```

Comment:
I would check that we can set resumeStatusCallback_ only if !isServer_ && isResumable_ and then on this place I would only check the resumeStatusCallback_ pointer.

Also it feels that since you received a frame, that means the transport layer is probably working just fine.
We should probably just message Resume failure.

The underlaying DuplexConnection might terminate with cancel, onComplete or onError. That is a transport layer failure which we should hook up to and message it as network failure.

BTW, does onConnectionError mean resume error?